Makefile

```diff
@@ -125,3 +125,25 @@ docs:
 .PHONY : clean-docs
 clean-docs:
 	$(MAKE) -C doc clean
+
+.PHONY : docker
+ifdef target
+    TARGET := $(target)
+else
+    TARGET := lightning-qubit
+endif
+docker-build:
+	docker build -f docker/Dockerfile --tag=pennylaneai/pennylane:$(TARGET) --target wheel-$(TARGET) .
```

Comment:
Would it be better to use `--tag=pennylaneai/pennylane-$(TARGET):${TAG}` or equivalent, so that we can version the images? Use `latest` for master/main and by default, and then 0.32.0, etc otherwise for fixed releases?

Reply:
The scope and specs on this story were not clear at the beginning, so I went for `stable` and L, L-GPU and L-Kokkos-OMP first. Since L-GPU worked I added L-Kokkos-CUDA, and since I had L-Kokkos-CUDA I thought I'll try adding L-Kokkos-ROCm to have all single-device backends. Now, I didn't know if the goal was to provide recipes that users can modify to their needs, or have a complete set of official images. If we go for the later, then your suggestion makes a lot of sense. Otherwise, still makes sense, but does not matter as much. I think maybe we should ask @trbromley what he thinks about the scope at this point.